addon/model.js

```diff
@@ -43,17 +46,24 @@ class EmbeddedSnapshot {
   }
 
   eachAttribute(callback, binding) {
+    if (false) {
+      let recordData = recordDataFor(this.record);
+      return recordData.eachAttribute(callback, binding);
+    }
     return this.record.eachAttribute(callback, binding);
   }
 
   attr(key) {
     return this.attrs[key];
   }
 }
-
 // TODO: shouldn't need this anymore; this level of indirection for nested recordData isn't useful
 export class EmbeddedInternalModel {
```

Comment:
Let's move this class to `addon/resolve-attribute-util` and not define the class at all in the custom model class case

Reply:
moved